src/domain/categories.ts

```diff
@@ -45,46 +42,81 @@ export function editCategory(props: Category) {
 
 //ini namanya custom hooks
 export function useFetchCategories() {
-  const [categories, setCategories] = useState<Category[]>();
+  const [categories, setCategories] = useState<Category[]>([]);
+  const [errorMessageFetch, seterrorMessageFetch] = useState<string>("");
   const [isLoadingFetch, setIsLoadingFetch] = useState(false);
   useEffect(() => {
-    fetchData().then((categories) => {
-      setCategories(categories);
-    });
+    setIsLoadingFetch(true);
+    fetchData()
+      .then((data: Category[]) => {
+        setCategories([...data]);
+        setIsLoadingFetch(false);
+      })
+      .catch((err) => {
+        setCategories([]);
+        seterrorMessageFetch(`${err.message} + error in get data`);
+        setIsLoadingFetch(false);
+      });
   }, []);
   function reFetch() {
-    fetchData().then((categories) => {
-      setCategories(categories);
-    });
+    setIsLoadingFetch(true);
+    fetchData()
+      .then((data: Category[]) => {
+        setCategories([...data]);
+        setIsLoadingFetch(false);
+      })
+      .catch((err) => {
+        setCategories([]);
+        seterrorMessageFetch(`${err.message} + error in get data`);
+        setIsLoadingFetch(false);
+      });
```

Comment:
Ini juga bisa diupdate biar lebih rapi, pake finally

```suggestion
    fetchData()
      .then((data) => {
        setCategories(data);
        setErrorMessageFetch("");
      })
      .catch((err) => {
        setCategories([]);
        setErrorMessageFetch(`${err.message} + error in get data`);
      })
      .finally(() => setIsLoadingFetch(false));
```

Reply:
done update mas